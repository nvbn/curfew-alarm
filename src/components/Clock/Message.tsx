import React from "react";
import { Text } from "react-native-svg";

import i18n from "../../utils/i18n";
import {
  Status,
  STATUS_FINE,
  STATUS_GO_HOME_WHEN_CURFEW,
  STATUS_GO_HOME_WHEN_TIME_TO_GO_HOME,
  STATUS_STAY_AT_HOME_WHEN_CURFEW,
  STATUS_STAY_AT_HOME_WHEN_TIME_TO_GO_HOME,
} from "../../utils/status";
import { COLOR_CURFEW, COLOR_DANGER, COLOR_OK } from "./styles";

type BaseMessageProps = {
  text: string;
  color: string;
};

const BaseMessage = ({ text, color }: BaseMessageProps): JSX.Element => (
  <Text x={50} y={60} fill={color} textAnchor="middle" fontSize={6}>
    {text}
  </Text>
);

type Props = {
  status: Status;
};

/**
 * A message in SVG with the action expected from a user.
 */
const Message = ({ status }: Props): JSX.Element => {
  switch (status) {
    case STATUS_FINE:
      return <BaseMessage text={i18n.t("clockMessageFine")} color={COLOR_OK} />;
    case STATUS_GO_HOME_WHEN_TIME_TO_GO_HOME:
      return (
        <BaseMessage
          text={i18n.t("clockMessageTimeToGoHome")}
          color={COLOR_DANGER}
        />
      );
    case STATUS_STAY_AT_HOME_WHEN_TIME_TO_GO_HOME:
      return (
        <BaseMessage
          text={i18n.t("clockMessageStayAtHome")}
          color={COLOR_DANGER}
        />
      );
    case STATUS_GO_HOME_WHEN_CURFEW:
      return (
        <BaseMessage text={i18n.t("clockMessageGoHome")} color={COLOR_CURFEW} />
      );
    case STATUS_STAY_AT_HOME_WHEN_CURFEW:
      return (
        <BaseMessage
          text={i18n.t("clockMessageStayAtHome")}
          color={COLOR_CURFEW}
        />
      );
  }
};

export default Message;
