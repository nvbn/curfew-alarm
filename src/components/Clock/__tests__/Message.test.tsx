import { render } from "@testing-library/react-native";
import React from "react";

import {
  Status,
  STATUS_FINE,
  STATUS_GO_HOME_WHEN_CURFEW,
  STATUS_GO_HOME_WHEN_TIME_TO_GO_HOME,
  STATUS_STAY_AT_HOME_WHEN_CURFEW,
  STATUS_STAY_AT_HOME_WHEN_TIME_TO_GO_HOME,
} from "../../../utils/status";
import Message from "../Message";

describe("<Message />", () => {
  for (const status of [
    STATUS_FINE,
    STATUS_GO_HOME_WHEN_CURFEW,
    STATUS_GO_HOME_WHEN_TIME_TO_GO_HOME,
    STATUS_STAY_AT_HOME_WHEN_CURFEW,
    STATUS_STAY_AT_HOME_WHEN_TIME_TO_GO_HOME,
  ] as Status[]) {
    test(`when ${status}`, () => {
      const rendered = render(<Message status={status} />);

      expect(rendered.toJSON()).toMatchSnapshot();
    });
  }
});
