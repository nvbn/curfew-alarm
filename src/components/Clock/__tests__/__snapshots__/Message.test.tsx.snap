// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> when FINE 1`] = `
<RNSVGText
  content={null}
  dx={Array []}
  dy={Array []}
  fill={4281589372}
  font={
    Object {
      "fontSize": 6,
      "textAnchor": "middle",
    }
  }
  propList={
    Array [
      "fill",
    ]
  }
  rotate={Array []}
  x={
    Array [
      50,
    ]
  }
  y={
    Array [
      60,
    ]
  }
>
  <RNSVGTSpan
    content="You're fine"
    dx={Array []}
    dy={Array []}
    font={Object {}}
    rotate={Array []}
    x={Array []}
    y={Array []}
  />
</RNSVGText>
`;

exports[`<Message /> when GO_HOME_WHEN_CURFEW 1`] = `
<RNSVGText
  content={null}
  dx={Array []}
  dy={Array []}
  fill={4292763476}
  font={
    Object {
      "fontSize": 6,
      "textAnchor": "middle",
    }
  }
  propList={
    Array [
      "fill",
    ]
  }
  rotate={Array []}
  x={
    Array [
      50,
    ]
  }
  y={
    Array [
      60,
    ]
  }
>
  <RNSVGTSpan
    content="Go home!"
    dx={Array []}
    dy={Array []}
    font={Object {}}
    rotate={Array []}
    x={Array []}
    y={Array []}
  />
</RNSVGText>
`;

exports[`<Message /> when GO_HOME_WHEN_TIME_TO_GO_HOME 1`] = `
<RNSVGText
  content={null}
  dx={Array []}
  dy={Array []}
  fill={4292788264}
  font={
    Object {
      "fontSize": 6,
      "textAnchor": "middle",
    }
  }
  propList={
    Array [
      "fill",
    ]
  }
  rotate={Array []}
  x={
    Array [
      50,
    ]
  }
  y={
    Array [
      60,
    ]
  }
>
  <RNSVGTSpan
    content="Time to go home"
    dx={Array []}
    dy={Array []}
    font={Object {}}
    rotate={Array []}
    x={Array []}
    y={Array []}
  />
</RNSVGText>
`;

exports[`<Message /> when STAY_AT_HOME_WHEN_CURFEW 1`] = `
<RNSVGText
  content={null}
  dx={Array []}
  dy={Array []}
  fill={4292763476}
  font={
    Object {
      "fontSize": 6,
      "textAnchor": "middle",
    }
  }
  propList={
    Array [
      "fill",
    ]
  }
  rotate={Array []}
  x={
    Array [
      50,
    ]
  }
  y={
    Array [
      60,
    ]
  }
>
  <RNSVGTSpan
    content="Stay at home"
    dx={Array []}
    dy={Array []}
    font={Object {}}
    rotate={Array []}
    x={Array []}
    y={Array []}
  />
</RNSVGText>
`;

exports[`<Message /> when STAY_AT_HOME_WHEN_TIME_TO_GO_HOME 1`] = `
<RNSVGText
  content={null}
  dx={Array []}
  dy={Array []}
  fill={4292788264}
  font={
    Object {
      "fontSize": 6,
      "textAnchor": "middle",
    }
  }
  propList={
    Array [
      "fill",
    ]
  }
  rotate={Array []}
  x={
    Array [
      50,
    ]
  }
  y={
    Array [
      60,
    ]
  }
>
  <RNSVGTSpan
    content="Stay at home"
    dx={Array []}
    dy={Array []}
    font={Object {}}
    rotate={Array []}
    x={Array []}
    y={Array []}
  />
</RNSVGText>
`;
